### ISSUE TITLE
how to implement a tflite compatible SVDF layer

### ISSUE THREAD

**Issue by kismeter** [keras-team/keras#19962]
Hi,

Keras layers don't have SVDF layer that compatible with tflite SVDF ops, is there any official plan to implement this? if implement a customized tflite compatible SVDF layer, is there any how to guidance?

**Comment by mehtamansi29** [comment]
Hi @kismeter - 

There is open source [repo](https://github.com/google-research/google-research/blob/master/kws_streaming/layers/svdf.py) where custom SVDF layer class is created. You can use this SVDF class to create model. After that convert that model into TFLite which you can use it on mobile device. 
This [link](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper_12_labels.md#install-tensorflow-with-deps) is also helpful to understand and setup kws_streaming.

The steps required to performed:
• Design a model using Keras layers and Stream wrapper.
• Train the model.
• Automatically convert the trained non-streaming Keras model to Keras streaming one.
• Convert streaming Keras model to a TFLite module.Quantize it if needed to optimize model size and performance.
• Benchmark TFLite module: measure accuracy and latency on CPU of mobile phone Pixel4

For more details regarding tflite compatible SVDF layer, kindly raise an issue in tflite repo.Thanks...!!

**Comment by kismeter** [comment]
Hi @mehtamansi29 ,

I already tried the open source custom SVDF layer, that implement can't fuse to TFLite SVDF Ops, they avoid to using TFLite  custom SVDF Ops from rybakov [comments](https://github.com/google-research/google-research/issues/1950#issuecomment-2002573529), so just want to know if keras has plan to implement tflite compatible SVDF layer

**Comment by mehtamansi29** [comment]
Hi @kismeter -

TensorFlow Lite SVDF operations is experimental operation but not ready for custom models. For single value decomposition of matrics [tf.keras.ops.svd](https://www.tensorflow.org/api_docs/python/tf/keras/ops/svd) and [tf.raw_ops.Svd](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Svd) is there.

**Comment by divyashreepathihalli** [comment]
Hi @mehtamansi29, The keras team doesnt plan on adding support for this layer. However, someone has tried to do this before - here is a link to that issue - https://github.com/tensorflow/tensorflow/issues/63035
